product_pricelist_extended/models/product_pricelist_item.py

```diff
@@ -23,35 +24,108 @@ class ProductPricelistItem(models.Model):
         column1="pricelist_item_id",
         column2="tmpl_id",
         string="Templates",
-        oldname="tmpl_ids",
     )
     price_categ_id = fields.Many2one(
-        comodel_name="product.price.category",
-        string="Pricing Category",
-        oldname="price_categ",
+        comodel_name="product.price.category", string="Pricing Category"
     )
 
     code_inclusion = fields.Char("Code includes")
     code_exclusion = fields.Char("Code excludes")
 
-    # Inherits upstream usage of deprecated api.one
-    # pylint: disable=W8104
-    @api.one
     @api.depends("price_categ_id", "product_tmpl_ids", "product_ids")
     def _get_pricelist_item_name_price(self):
         super()._get_pricelist_item_name_price()
-        if self.price_categ_id:
-            self.name = _("Price Category: %s") % self.price_categ_id.name
-        elif self.product_tmpl_ids:
-            self.name = ",".join(self.product_tmpl_ids.mapped("name"))
-        elif self.product_ids:
-            self.name = ",".join(
-                [p.display_name.replace("[%s]" % p.code, "") for p in self.product_ids]
-            )
-        suffix = []
-        if self.code_inclusion:
-            suffix.append(_("contains %s") % self.code_inclusion)
-        if self.code_exclusion:
-            suffix.append(_("excludes %s") % self.code_exclusion)
-        if suffix:
-            self.name = self.name + _("(Code %s)") % ", ".join(suffix)
+        for item in self:
+            if item.price_categ_id and item.applied_on == "2_product_category":
+                item.name = _("Price Category: %s") % item.price_categ_id.display_name
+            elif item.product_tmpl_ids and item.applied_on == "1_product":
+                # odoo creates new objects for item.product_tmpl_ids
+                # but this doesn't happen in v12
+                item.name = _("Product: %s") % (
+                    ",".join(item.product_tmpl_ids.mapped("display_name"))
+                )
+            elif item.product_ids and item.applied_on == "0_product_variant":
+                # odoo creates new objects for item.product_ids, so I put _origin here
```

Comment:
This really doesn't look correct. The newid mechanism was introduced in v10, it isn't new, so if we are seeing it here something else is going on. Even within an onchange, the product objects shoud be originals. We really shouldn't be using origin here unless the upstream function is doing the same..